src/Ellipsoid.jl

```diff
@@ -231,3 +231,32 @@ function rand(::Type{Ellipsoid};
                    Matrix{N}(dim*I, dim, dim)
     return Ellipsoid(center, shape_matrix)
 end
+
+"""
+    translate(E::Ellipsoid{N}, v::AbstractVector{N}) where {N<:AbstractFloat}
+
+Translate (i.e., shift) an ellipsoid by a given vector.
+
+### Input
+
+- `E` -- ellipsoid
+- `v` -- translation vector
+
+### Output
+
+A translated ellipsoid.
+
+### Notes
+
+The shape matrix is shared with the original ellipsoid.
+
+### Algorithm
+
+We add the vector to the center of the ellipsoid.
+"""
+function translate(E::Ellipsoid{N},
+                   v::AbstractVector{N}) where {N<:AbstractFloat}
+    @assert length(v) == dim(E) "cannot translate a $(dim(E))-dimensional " *
+                                "set by a $(length(v))-dimensional vector"
+    return Ellipsoid(center(E) + v, E.shape_matrix)
```

Comment:
This choice doesn't convince me -- why share the shape matrix and not the center? 

Have you considered adding a `translate!` method (in addition to `translate`, which returns a new object), which shares both center and shape matrix?

Reply:
> why share the shape matrix

The sharing is for efficiency. Normally you do not modify a set in-place, so it is safe to share structure.

> and not the center

Because the center would be modified.

> Have you considered adding a `translate!` method

I thought about it, but the object is not actually modified, so the `!` is not correct.
Maybe we want to have an additional optional argument (something like be `share` or `copy`)?

Reply:
My proposal was to either (i) make `translate` to return a new independent object or (ii) make translate return a new independent object *and* also offer `translate!` for performance, which translates the object in-place. The implementation is as follows:

```julia
function translate(E::Ellipsoid{N},
                   v::AbstractVector{N}) where {N<:AbstractFloat}
    @assert length(v) == dim(E) "cannot translate a $(dim(E))-dimensional " *
                                "set by a $(length(v))-dimensional vector"
    return translate!(copy(E), v)
end

function translate!(E::Ellipsoid{N},
                   v::AbstractVector{N}) where {N<:AbstractFloat}
    E.center .+= v
    return E
end
```

Reply:
And I do not like that because this does not give you the possibility to share things.

Reply:
To share things? Can you elaborate?

The behavior of `translate / translate!` of above is:

```julia
julia> E = Ellipsoid(zeros(2), Diagonal([1.0, 2.0]))
Ellipsoid{Float64}([0.0, 0.0], [1.0 0.0; 0.0 2.0])

julia> E_right = translate(E, [1.0, 0.0])
Ellipsoid{Float64}([1.0, 0.0], [1.0 0.0; 0.0 2.0])

julia> E_right.shape_matrix === E.shape_matrix
false

julia> E_right.center === E.center
false

julia> E_right_inplace = translate!(E, [1.0, 0.0])

julia> E_right_inplace.shape_matrix === E.shape_matrix
true

julia> E_right_inplace.center === E.center
true
```

Reply:
Applying `translate!` modifies `E.center`.
I want a way to share `E.shape_matrix` but to copy `E.center`. This is the minimal amount of copying (in order to not modify `E`) and the maximal amount of sharing (i.e., the unmodified part is shared), which is not possible with your proposal.